src/service/expenses.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+let expenses = [];
+
+const reset = () => {
+  expenses = [];
+};
+
+const filterExpanses = (expense, queryParams) => {
+  const { userId, categories, from, to } = queryParams;
+
+  return (
+    (!userId || expense.userId === Number(userId))
+    && (!categories || categories.includes(expense.category))
+    && (!from || expense.spentAt >= from)
+    && (!to || expense.spentAt <= to)
```

Comment:
Use `new Date(expense.spentAt) >= new Date(from)` instead of `expense.spentAt >= from` to ensure proper date comparisons.